### PR TITLE
DBZ-1297 Expand JSON escaped string as JSON

### PR DESCRIPTION
**Pull request by lbroudoux**
This is a first step of [DBZ-1297](https://issues.redhat.com/browse/DBZ-1297) for managing transformation of an escaped JSON string into JSON, inferring the connect Schema from the JSON document structure. 

This allow the Outbox Event Router to directly produce JSON message on Kafka like:

```json
# Kafka Topic: outbox.event.order
# Kafka Message key: "1"
# Kafka Message Headers: "id=4d47e190-0402-4048-bc2c-89dd54343cdc"
# Kafka Message Timestamp: 1556890294484
{
  "id": 1, "lineItems": [{"id": 1, "item": "Debezium in Action", "status": "ENTERED", "quantity": 2, "totalPrice": 39.98}, {"id": 2, "item": "Debezium for Dummies", "status": "ENTERED", "quantity": 1, "totalPrice": 29.99}], "orderDate": "2019-01-31T12:13:01", "customerId": 123
}
```

This option is off by default and can be turned-on using: `transforms.outbox.debezium.expand.json.payload=true`

### CODE REVIEW

**Comment by jpechane** [comment]
@lbroudoux  We are near the release of 1.7.0.Final. The PR will be merged right after the realse into 1.8 master.

File: debezium-core/src/test/java/io/debezium/transforms/outbox/EventRouterTest.java [hunk 1]
```diff
@@ -897,6 +897,65 @@ public void noTombstoneIfNotConfigured() {
...
+                EventRouterConfigDefinition.EXPAND_JSON_PAYLOAD.name(),
+                "true");
+        router.configure(config);
+
+        final SourceRecord eventRecord = createEventRecord(
+                "da8d6de6-3b77-45ff-8f44-57db55a7a06c",
+                "UserCreated",
+                "10711fa5",
+                "User",
+                "{\"fullName\": \"John Doe\", \"rating\": 4.9, \"age\": 42, \"pets\": [\"dog\", \"cat\"]}",
```
Comment (gunnarmorling) [inline debezium-core/src/test/java/io/debezium/transforms/outbox/EventRouterTest.java:914]: Could you add a `boolean` field, too?

**Review by gunnarmorling — APPROVED** [review]
Great work, thanks a lot, @lbroudoux! One small suggesting in regards to testing inline.

File: debezium-core/src/test/java/io/debezium/transforms/outbox/EventRouterTest.java (hunk 1 above)
Comment (lbroudoux) [reply debezium-core/src/test/java/io/debezium/transforms/outbox/EventRouterTest.java]: Yes of course. Done!

**Comment by gunnarmorling** [comment]
And applied; thanks again, @lbroudoux, that's a great addition!

**Pull request merged**